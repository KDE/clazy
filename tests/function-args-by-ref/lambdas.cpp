struct Trivial
{
};

struct NonTrivial
{
    ~NonTrivial();
};

void test()
{
    auto l1 = [](const Trivial &) { return 0; };
    auto l2 = [](Trivial) { return 0; };
    auto l3 = [](const NonTrivial &) { return 0; };
    auto l4 = [](const NonTrivial&) { return 0; };
}
