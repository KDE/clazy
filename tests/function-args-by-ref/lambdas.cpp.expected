function-args-by-ref/lambdas.cpp:15:18: warning: test()::(anonymous class)::operator(): Missing reference on non-trivial type (NonTrivial) [-Wclazy-function-args-by-ref]
